app/src/main/java/org/sopt/sample/adapter/RepoAdapter.kt

```diff
@@ -0,0 +1,78 @@
+package org.sopt.sample.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import org.sopt.sample.R
+import org.sopt.sample.data.Repo
+import org.sopt.sample.databinding.LayoutGithubRepoBinding
+import org.sopt.sample.databinding.LayoutRecyclerTextBinding
+import org.w3c.dom.Text
+import java.lang.ClassCastException
+import java.lang.RuntimeException
+
+class RepoAdapter(context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private val inflater by lazy { LayoutInflater.from(context) }
+    private var repoList: List<Repo> = emptyList()
+
+    class RepoViewHolder(
+        private val binding: LayoutGithubRepoBinding,
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun onRepoBind(data: Repo) {
+            binding.imgGithub.setImageDrawable(binding.root.context.getDrawable(data.image))
+            binding.txtGithubRepoName.text = data.name
+            binding.txtGithubRepoAuthor.text = data.author
+        }
+    }
+
+    class TextViewHolder(
+        private val binding: LayoutRecyclerTextBinding,
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun onTextBind(data: Repo) {
+            binding.rowText.text = data.name
+        }
+    }
+
+    //getItemViewType의 리턴값 Int가 viewType으로 넘어온다.
+    //viewType으로 넘어오는 값에 따라 viewHolder 처리해주기!
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            Repo.TEXT_TYPE -> {
+                val binding = LayoutRecyclerTextBinding.inflate(inflater, parent, false)
+                TextViewHolder(binding)
+            }
+            Repo.REPO_TYPE -> {
+                val binding = LayoutGithubRepoBinding.inflate(inflater, parent, false)
+                RepoViewHolder(binding)
+            }
+            else -> {
+                throw ClassCastException("Unknown viewType Error")
+            }
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return repoList[position].type
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (repoList[position].type) {
+            Repo.REPO_TYPE -> {
+                (holder as RepoViewHolder).onRepoBind(repoList[position])
+            }
+            Repo.TEXT_TYPE -> {
+                (holder as TextViewHolder).onTextBind(repoList[position])
+            }
+        }
```

Comment:
```suggestion
        when (holder) {
            is  RepoViewHolder-> {
                holder.onRepoBind(repoList[position])
            }
            is  TextViewHolder-> {
                holder.onTextBind(repoList[position])
            }
        }
```
`when` 을 사용하면 `as`로 viewholder를 캐스팅하지 않아도 분기처리를 할 수 있습니다!
또, 해당 아이템의 type으로 구분하지 않아도
이미 파라미터 `holder: RecyclerView.ViewHolder` 안에 ViewHolder가 구분된 상태로 들어옵니다!!